test/API/H5_api_test.h

```diff
@@ -62,11 +70,66 @@ extern const char *test_path_prefix;
 
 #define UNUSED(o) (void)(o);
 
-#define H5_API_TEST_FILENAME_MAX_LENGTH 1024
-
 /* The maximum size of a dimension in an HDF5 dataspace as allowed
  * for this testing suite so as not to try to create too large
  * of a dataspace/datatype. */
 #define MAX_DIM_SIZE 16
 
-#endif
+#ifdef H5_HAVE_MULTITHREAD
+/* Individual API tests may be run serially or in multiple threads.
+ * If being run in multiple threads, this macro will define a handler
+ * named <test>_thread which creates and tears down the threads.
+ * 
+ * TBD: Note that it invokes another macro-defined routine...which sets up threadlocal info...
+ * */
+#define DECLARE_MT_API_TEST_FUNC_OUTER(func) \
```

Comment:
This feels like a very over-engineered macro solution to a situation which would probably be better handled by just having logic in the testing framework itself to manage threads. I'd suggest that we consider a different approach where an individual test or the whole test program can signal that it would prefer for the testing framework to manage threads.

Reply:
Would it be reasonable to add a new parameter to `AddTest` to control whether a test is performed in multiple threads? Since the test parameters are opaque to the test framework, I don't think there are many other reasonable ways to do it. We could maybe check the test name for an `mt_` prefix or something like that, but that seems opaque and easy to forget when adding new tests later on.

Reply:
I'm thinking something like that, yes, though maybe give some consideration towards if there is a more generic way to extend `AddTest` so that we can reduce the need to go and update calls to it everywhere whenever we need a new testing parameter. I think it will simply be more maintainable to have the testing framework handle threads when asked for, rather than using an ad-hoc macro-based solution. Any time you find that you're reaching for macros to basically implement an entire function, it should usually tell you right away that there's likely a better solution.